core/serialization/variables.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Handles serializing variables to plain JavaScript objects, only
+ *     containing state.
+ */
+'use strict';
+
+goog.module('Blockly.serialization.variables');
+goog.module.declareLegacyNamespace();
+
+const Events = goog.require('Blockly.Events');
+// eslint-disable-next-line no-unused-vars
+const IPluginSerializer =
+    goog.requireType('Blockly.serialization.IPluginSerializer');
+// eslint-disable-next-line no-unused-vars
+const Workspace = goog.requireType('Blockly.Workspace');
+const priorities = goog.require('Blockly.serialization.priorities');
+const registry = goog.require('Blockly.registry');
+
+
+// TODO: Is there any way to not export these? I would rather keep this out of
+//     the public API if possible.
+
+/**
+ * Represents the state of a given variable.
+ * @typedef {{
+ *   name: string,
+ *   id: string,
+ *   type: ?string,
+ * }}
+ */
+// eslint-disable-next-line no-unused-vars
+var State;
+exports.State = State;
+
+/**
+ * Returns the state of the variable as a plain JavaScript object.
+ * @param {!Blockly.variableModel} variableModel The variable to serialize.
+ * @return {!Blockly.serialization.variables.State} The serialized state of the
+ *     variable.
+ */
+const save = function(variableModel) {
+  const state = Object.create(null);
+  state['name'] = variableModel.name;
+  state['id'] = variableModel.getId();
+  if (variableModel.type) {
+    state['type'] = variableModel.type;
+  }
+  return state;
+};
+exports.save = save;
```

Comment:
I think I can stop exporting this now, because it's wrapped in the plugin serializer. This is perfect because I didn't want these to be public anyway.